### ISSUE TITLE
Legend location in figure with multiple axes

### ISSUE THREAD

**Issue by cmoscardi** [geopandas/geopandas#630]
Hi, when attempting to run code like this:

```
fig, axes = plt.subplots(2, 2, figsize=(16, 20))
axes = axes.flatten()
cols_to_plot = ["col_1", "col_2", "col_3", "col_4"]
for ax, col in zip(axes, cols_to_plot):
    df.plot(col, ax=ax, legend=True)
```

The colorbars for all 4 plots appear in the bottom right of the figure.

I'm not a matplotlib expert, but I believe the offending line of code is here:
https://github.com/geopandas/geopandas/blob/master/geopandas/plotting.py#L494

Perhaps worth remarking that I'm on a Jupyter notebook when I try this.

**Comment by jorisvandenbossche** [comment]
This might be the same issue as https://github.com/geopandas/geopandas/issues/522, which is fixed on master in the meantime.

**Comment by ljwolf** [comment]
I think I submitted a fix to this in #523

**Comment by cmoscardi** [comment]
Ah great, that looks like it. I'll try the latest master. Thank you!
